File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
    let answer = 0;
    const n = numbers.length;
    
    const dfs = (sum, index) => {
        if (index === n) {
            if (sum === target) answer++;
            return;
        }
        
        dfs(sum + numbers[index], index + 1);
        dfs(sum - numbers[index], index + 1);
    }
    
    dfs(0, 0);
    
    return answer;
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    answer = 0
    n = len(numbers)
    
    def dfs(_sum, _index):
        nonlocal answer
        
        if _index == n:
            if _sum == target: answer += 1
            return
        
        dfs(_sum + numbers[_index], _index + 1)
        dfs(_sum - numbers[_index], _index + 1)
        
    dfs(0, 0)
    
    return answer
